src/test/pybind/test_rados.py

```diff
@@ -583,6 +583,27 @@ def test_clear_omap(self):
             self.ioctx.operate_read_op(read_op, "hw")
             eq(list(iter), [])
 
+    def test_xattrs_op(self):
+        xattrs = dict(a=b'1', b=b'2', c=b'3', d=b'a\0b', e=b'\0')
+        with WriteOpCtx() as write_op:
+            write_op.new(LIBRADOS_CREATE_EXCLUSIVE)
+            for key, value in xattrs.items():
+                write_op.setxattr(key, value)
```

Comment:
think you are not testing set_xattr method, can you update and paste test run?

Reply:
@ideepika udpated,please check. thanks.
test code:
```
from rados import (Rados, WriteOpCtx, LIBRADOS_CREATE_EXCLUSIVE)
from nose.tools import eq_ as eq, ok_ as ok, assert_raises
import threading
import rados
import unittest

class TestIoctx(unittest.TestCase):
    def setUp(self):
        self.rados = Rados(conffile='./ceph.conf')
        self.rados.connect()
        self.rados.create_pool('poll10')
        assert self.rados.pool_exists('poll10')
        self.ioctx = self.rados.open_ioctx('poll10')
    def tearDown(self):
        self.ioctx.close()
        self.rados.delete_pool('poll10')
        self.rados.shutdown()

    def test_xattrs_op(self):
        xattrs = dict(a=b'1', b=b'2', c=b'3', d=b'a\0b', e=b'\0')
        with WriteOpCtx() as write_op:
            write_op.new(LIBRADOS_CREATE_EXCLUSIVE)
            for key, value in xattrs.items():
                write_op.set_xattr(key, value)
                self.ioctx.operate_write_op(write_op, "abc")
                eq(self.ioctx.get_xattr('abc', key), value)
            stored_xattrs_1 = {}
            for key, value in self.ioctx.get_xattrs('abc'):
                stored_xattrs_1[key] = value
            eq(stored_xattrs_1, xattrs)
            for key in xattrs.keys():
                write_op.rm_xattr(key)
                self.ioctx.operate_write_op(write_op, "abc")
            stored_xattrs_2 = {}
            for key, value in self.ioctx.get_xattrs('abc'):
                stored_xattrs_2[key] = value
            eq(stored_xattrs_2, {})
            write_op.remove()

if __name__ == '__main__':
    unittest.main()
```
test result:
```
[root@jiaowa build]# python test_xattr.py
.
----------------------------------------------------------------------
Ran 1 test in 2.316s

OK
```